### PR TITLE
:seedling: Unit tests dependency_update_tool

### PR DESCRIPTION
**Pull request by naveensrinivasan**
Unit tests dependency_update_tool
https://github.com/ossf/scorecard/issues/986

* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: checks/evaluation/dependency_update_tool_test.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			got := DependencyUpdateTool(tt.args.name, tt.args.dl, tt.args.r)
+			if tt.want.Score != got.Score {
+				t.Errorf("DependencyUpdateTool() got Score = %v, want %v for %v", got.Score, tt.want.Score, tt.name)
+			}
+			if tt.err && got.Error2 == nil {
+				t.Errorf("DependencyUpdateTool() error = %v, want %v for %v", got.Error2, tt.want.Error2, tt.name)
```
Comment (laurentsimon) [inline checks/evaluation/dependency_update_tool_test.go:145]: can you add a verification of Warn/Info messages?

Comment (naveensrinivasan) [reply checks/evaluation/dependency_update_tool_test.go:171]: Yes, I included that. PTAL. Thanks

File: checks/evaluation/dependency_update_tool_test.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+				},
+			},
+			want: checker.CheckResult{
+				Score:  10,
+				Error2: nil,
+			},
+			expected: scut.TestReturn{
+				Error:        nil,
+				Score:        10,
+				NumberOfInfo: 1,
```
Comment (laurentsimon) [inline checks/evaluation/dependency_update_tool_test.go:113]: you may set the other `NNumberOfWarn` and debug to 0 explicitly. I'm not sure what's best :-)

**Review by laurentsimon — APPROVED** [review]
Thanks!

**Pull request merged**